src/common/multiline-ellipsis/index.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * Adds an ellipsis to a el based on the height and width
+ * specified in CSS rules
+ */
+function truncate(el) {
+    if (!el) {
+        return;
+    }
+
+    const text = el.innerText;
+
+    if (this.isTextOverflowing(el)) {
+        const checkFunc = i => {
+            el.innerText = text.substring(0, i);
+            return this.isTextOverflowing(el) ? -1 : 0;
+        };
+        const len = this.binarySearch(text.length - 1, checkFunc);
+        const ellipsisAmount = 2;
+        const truncatedText = text.substring(0, len).slice(0, -ellipsisAmount);
+        el.innerHTML = `<span aria-hidden="true">${truncatedText}…</span><span class="clipped">${text}</span>`;
+    }
+}
+
+function isTextOverflowing(el) {
+    const currentOverflow = el.style.overflow;
+    if (!currentOverflow || currentOverflow === 'visible') {
+        el.style.overflow = 'hidden';
+    }
+
+    const isOverflowing = el.clientWidth < el.scrollWidth || el.clientHeight < el.scrollHeight;
+    el.style.overflow = currentOverflow;
+    return isOverflowing;
+}
+
+function binarySearch(length, callback) {
+    let low = 0;
+    let high = length - 1;
+    let best = -1;
+    let mid;
+
+    while (low <= high) {
+        mid = ~~((low + high) / 2);
```

Comment:
For consistency with the binary search in the carousel code and for general readability we should probably change `~~` to `Math.floor(`.

Reply:
Sure.